code/modules/cm_marines/overwatch.dm

```diff
@@ -57,437 +70,343 @@
+		if(!marine)
+			continue //just to be safe
+		var/mob_name = "unknown"
+		var/mob_state = ""
+		var/has_helmet = TRUE
+		var/role = "unknown"
+		var/acting_sl = ""
+		var/fteam = ""
+		var/distance = "???"
+		var/area_name = "???"
+		var/is_squad_leader = FALSE
+		var/mob/living/carbon/human/marine_human
+
+
+		if(ishuman(marine))
+			marine_human = marine
+			if(istype(marine_human.loc, /obj/structure/machinery/cryopod)) //We don't care much for these
+				continue
+			mob_name = marine_human.real_name
+			var/area/current_area = get_area(marine_human)
+			var/turf/current_turf = get_turf(marine_human)
+			if(!current_turf)
+				continue
+			if(current_area)
+				area_name = sanitize_area(current_area.name)
+
+			switch(z_hidden)
+				if(HIDE_ALMAYER)
+					if(is_mainship_level(current_turf.z))
+						continue
+				if(HIDE_GROUND)
+					if(is_ground_level(current_turf.z))
+						continue
+
+			if(marine_human.job)
+				role = marine_human.job
+			else if(istype(marine_human.wear_id, /obj/item/card/id)) //decapitated marine is mindless,
+				var/obj/item/card/id/ID = marine_human.wear_id //we use their ID to get their role.
+				if(ID.rank)
+					role = ID.rank
+
+
+			if(current_squad.squad_leader)
+				if(marine_human == current_squad.squad_leader)
+					distance = "N/A"
+					if(current_squad.name == SQUAD_SOF)
+						if(marine_human.job == JOB_MARINE_RAIDER_CMD)
+							acting_sl = " (direct command)"
+						else if(marine_human.job != JOB_MARINE_RAIDER_SL)
+							acting_sl = " (acting TL)"
+					else if(marine_human.job != JOB_SQUAD_LEADER)
+						acting_sl = " (acting SL)"
+					is_squad_leader = TRUE
+				else if(current_turf && (current_turf.z == SL_z))
+					distance = "[get_dist(marine_human, current_squad.squad_leader)] ([dir2text_short(get_dir(current_squad.squad_leader, marine_human))])"
+
+
+			switch(marine_human.stat)
+				if(CONSCIOUS)
+					mob_state = "Conscious"
+
+				if(UNCONSCIOUS)
+					mob_state = "Unconscious"
+
+				if(DEAD)
+					mob_state = "Dead"
+
+			if(!istype(marine_human.head, /obj/item/clothing/head/helmet/marine))
+				has_helmet = FALSE
+
+			if(!marine_human.key || !marine_human.client)
+				if(marine_human.stat != DEAD)
+					mob_state += " (SSD)"
+
+
+			if(marine_human.assigned_fireteam)
+				fteam = " [marine_human.assigned_fireteam]"
+
+		else //listed marine was deleted or gibbed, all we have is their name
+			for(var/datum/data/record/marine_record as anything in GLOB.data_core.general)
+				if(marine_record.fields["name"] == marine)
+					role = marine_record.fields["real_rank"]
+					break
+			mob_state = "Dead"
+			mob_name = marine
+
+
+		switch(role)
+			if(JOB_SQUAD_LEADER)
+				leader_count++
+				if(mob_state != "Dead")
+					leaders_alive++
+			if(JOB_SQUAD_TEAM_LEADER)
+				ftl_count++
+				if(mob_state != "Dead")
+					ftl_alive++
+			if(JOB_SQUAD_SPECIALIST)
+				spec_count++
+				if(istype(marine_human.wear_id, /obj/item/card/id)) //decapitated marine is mindless,
```

Comment:
this can be null

```
runtime error: Cannot read null.wear_id
proc name: ui data (/obj/structure/machinery/computer/overwatch/ui_data)
  source file: code/modules/cm_marines/overwatch.dm,252
  usr: Brandon Quiri (/mob/dead/observer)
  src: Overwatch Console (/obj/structure/machinery/computer/overwatch/almayer)
  usr.loc: the floor (94,154,3) (/turf/open/floor/almayer)
  src.loc: the floor (92,151,3) (/turf/open/floor/almayer)
  call stack:
  ```